File: FIle Templates/Complete Custom View.xctemplate/UIViewObjective-C/___FILEBASENAME___.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import "___FILEBASENAME___.h"

@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - Lifecycle

- (instancetype)initWithCoder:(NSCoder *)aDecoder
{
    if (self = [super initWithCoder:aDecoder]) {
        [self commonInit];
    }
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        [self commonInit];
    }
    return self;
}

/**
 Called at init time.
 
 Call [super commonInit] when use this method at inheritance.
 */
- (void)commonInit
{
    [self loadViewFromNib];
    
    [self.contentView setFrame:self.bounds];
    [self.contentView setBackgroundColor:[UIColor clearColor]];
    
    [self addSubview:self.contentView];
}

#pragma mark - Custom Accessors


#pragma mark - IBActions


#pragma mark - Public


#pragma mark - Private

/**
 Loads view from Nib, considering that nibName is the same as class name.
 */
- (void)loadViewFromNib
{
    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
    UINib *nib = [UINib nibWithNibName:NSStringFromClass([self class]) bundle:bundle];
    
    [nib instantiateWithOwner:self options:nil];
}

@end

File: FIle Templates/Base Custom View.xctemplate/UIViewSwift/___FILEBASENAME___.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import UIKit

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {
    
    /// If you want to customize this view by simply adding additional views, you should add them to the content view
    @IBOutlet weak private(set) var contentView: UIView!
    
    // MARK: - Lifecycle
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        commonInit()
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }
    
    /**
    Called at init time.
    
    Call super.commonInit() when use this method at inheritance.
    */
    func commonInit() {
        loadViewFromNib()
        
        contentView.frame = bounds
        contentView.backgroundColor = UIColor.clearColor()
        
        addSubview(contentView)
    }
    
    /**
    Loads view from Nib, considering that nibName is the same as class name.
    */
    func loadViewFromNib() {
        let nibName = String(self.dynamicType)
        let bundle  = NSBundle(forClass: self.dynamicType)
        let nib     = UINib(nibName: nibName, bundle: bundle)
        nib.instantiateWithOwner(self, options: nil)
    }
}
